src/main/java/org/broadinstitute/hellbender/tools/copynumber/gcnv/GenerateVCFFromPosteriors.java

```diff
@@ -0,0 +1,224 @@
+                new SimpleIntervalCollection(new File(getIntervalFileFromChunkDirectory(chunkRootDirectory.getAbsolutePath()).getAbsolutePath()));
+
+        //check that sequence dictionaries are equivalent among all chunks
+        if (!chunkSimpleIntervalCollection.getMetadata().getSequenceDictionary().equals(samSequenceDictionary)) {
+            throw new UserException.BadInput("Not all sequence dictionaries are equivalent among interval lists in chunk directories");
+        }
+        final File chunkPosteriorFile = getPosteriorFileFromChunkDirectory(chunkRootDirectory.getAbsolutePath(), sampleDirectoryName);
+
+        final List<String> copyNumberStateColumns = getPosteriorFileColumns(chunkPosteriorFile);
+        final IntegerCopyNumberStateCollection chunkCopyNumberStateCollection = new IntegerCopyNumberStateCollection(copyNumberStateColumns);
+
+        final ChunkedCopyNumberPosteriorCollection chunkedCopyNumberPosteriorCollection =
+                new ChunkedCopyNumberPosteriorCollection(chunkPosteriorFile, chunkCopyNumberStateCollection);
+
+        //Combine together chunked interval list and chunked posterior collection into list of CopyNumberPosteriorLocatableRecord
+        final List<CopyNumberPosteriorLocatableRecord> copyNumberPosteriorLocatableRecordList =
+                IntStream.range(0, chunkedCopyNumberPosteriorCollection.size())
+                        .mapToObj(i -> new CopyNumberPosteriorLocatableRecord(chunkSimpleIntervalCollection.getIntervals().get(i),
+                                chunkedCopyNumberPosteriorCollection.getRecords().get(i)))
+                        .collect(Collectors.toList());
+
+        return new ImmutableTriple<>(copyNumberPosteriorLocatableRecordList, chunkCopyNumberStateCollection, chunkedCopyNumberPosteriorCollection.getMetadata().getSampleName());
+    }
+
+    /**
+     * Get list of column names of the copy number posterior file
+     */
+    private static List<String> getPosteriorFileColumns(final File copyNumberPosteriorFile) {
+        List<String> columns = null;
+        try (final XReadLines reader = new XReadLines(copyNumberPosteriorFile)) {
+            while (reader.hasNext()) {
+                String nextLine = reader.next();
+                if (!nextLine.startsWith(COMMENT_PREFIX)) {
+                    columns = Arrays.asList(nextLine.split(TableUtils.COLUMN_SEPARATOR_STRING));
+                    break;
+                }
+            }
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(copyNumberPosteriorFile);
+        }
+        if (columns == null) {
+            throw new UserException.BadInput("Copy number posterior file does not have a header");
+        }
+        return columns;
+    }
+
+    /**
+     * Get the posterior file from chunk directory
+     * TODO extract this to a static utility class for parsing GCNV directory tree structure
```

Comment:
i think these are fine here, it's unlikely any other tool would need these methods.  you can remove this and the below TODO.

Reply:
Done